### PR TITLE
Add back support for logging in the gradient clipping hooks

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Reverts unintentional breaking change from https://github.com/Lightning-AI/lightning/pull/14069

Reported in Slack by @ericharper 

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

### CODE REVIEW

File: src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py [hunk 1]
```diff
@@ -44,8 +44,13 @@ class _LogOptions(TypedDict):
             allowed_on_step=(False, True), allowed_on_epoch=(False, True), default_on_step=True, default_on_epoch=False
         ),
         "lr_scheduler_step": None,
-        "configure_gradient_clipping": None,
-        "clip_gradients": None,
+        # should match `optimizer_step`
+        "configure_gradient_clipping": _LogOptions(
+            allowed_on_step=(False, True), allowed_on_epoch=(False, True), default_on_step=True, default_on_epoch=False
+        ),
+        "clip_gradients": _LogOptions(
```
Comment (carmocca) [inline src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py:51]: This hook is not meant to be overridden, so we technically shouldn't support `log`ging in it. However, since it's called directly from `configure_gradient_clipping`, we don't have a place to reliably disable logging in it.

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py (hunk 1 above)
Comment (rohitgr7) [reply src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py:51]: I don't think we need to add logging for this one because this is called by user without any override and logging func name will remain `configure_gradient_clipping`.

Comment (rohitgr7) [reply src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py:51]: the reason I added this here is just for the sake of the test case.

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py (hunk 1 above)
Comment (carmocca) [reply src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py:51]: As I described in my previous message, I couldn't find a reliable way to disable logging for this hook. It would need to be done in the `configure_gradient_clipping` hook but it wouldn't work for users that override it

**Review by rohitgr7 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
